versions.bzl

```diff
@@ -0,0 +1 @@
+VERSION = "1.19.0-dev"
```

Comment:
How this will be used? I'm thinking whether this can be read from VERSION file instead of hardcoding.

Reply:
you cant use file *contents* in bazel `name` or `outs` fields, only variables set in bazel - that is why i want to add it - so we can have rules with names containing the version number

Reply:
relatedly - `pkg_deb` supports using a version file but `pkg_tar2rpm` doesnt

i opened a ticket here https://github.com/google/rpmpack/issues/63 regarding support for this

its possible i could get `pkg_rpm` to work instead of using `pkg_tar2rpm`

**however**, having a bazel version is still required for my current implementation which, in addition to creating `envoy` packages, creates packages for major versions too - eg `envoy-1.19` 

this allows users to install specific versions from the same repo

for that to work the version **must** be in bazel as its required for `names` and `outs` in the rules

apart from the versioned packaging, having the version in bazel allows you to inject it into titles, descriptions, etc etc

not having it in bazel means that apart from issues with specific rules - any time we need to mangle/inject something with the version we have to also depend on and open the file in a eg genrule - so it would be messy

if there was a way to parse the existing `VERSION` into some kind of bazel var that would be great but i didnt find any

using an `--action-env` wont work as it needs to be in the bazel rules not in *a* bazel rule

Reply:
> if there was a way to parse the existing VERSION into some kind of bazel var that would be great but i didnt find any

I think a `repository_rule` of bazel works, basically you can generate `versions.bzl` into a bazel repository (i.e. `@envoy_version//:versions.bzl` in that rule. Then you can refer that in `name` and `outs` or anywhere.

Refer https://github.com/envoyproxy/envoy/blob/main/bazel/dev_binding.bzl how to setup a repository rule.

Reply:
ok, thanks, ill try this and see if i can get it to work

Reply:
Yep, `repository_rule` should work here, good idea.

Reply:
updated to use a repository_rule